NDVI/NDVI.py

```diff
@@ -0,0 +1,361 @@
+# =============================================================================
+#  USGS/EROS Inventory Service Example
+#  Description: Download Landsat Collection 2 files
+#  Usage: python download_sample.py -u username -p password -f filetype
+#         optional argument f refers to filetype including 'bundle' or 'band'
+# =============================================================================
+
+import json
+import requests
+import sys
+import time
+import re
+import threading
+import os
+import numpy as np
+import csv
+import json
+import os
+import threading
+from datetime import datetime
+
+from landsatxplore.api import API
+
+path = "landsat-data"
+maxthreads = 5  # Threads count for downloads
+sema = threading.Semaphore(value=maxthreads)
+label = datetime.now().strftime("%Y%m%d_%H%M%S")  # Customized label using date time
+threads = []
+
+
+def get_credentials(path="usgs-pass.json"):
+    usgs_username = ""
+    usgs_password = ""
+    if os.path.exists(path):
+        with open(path, 'r') as file:
+            json_data = json.load(file)
+            usgs_username = json_data['username']
+            usgs_password = json_data['password']
+    else:
+        usgs_username = os.getenv("LANDSATXPLORE_USERNAME")
+        usgs_password = os.getenv("LANDSATXPLORE_PASSWORD")
+    if not usgs_username and not usgs_password:
+        print("No Credentials found.\n"
+              "Export LANDSATXPLORE_USERNAME and LANDSATXPLORE_PASSWORD")
+        return None
+    return usgs_username, usgs_password
+
+
+def get_last_date(path="/NDVI/ndvi.csv"):
+    """Get last recorded date from NDVI/ndvi.csv"""
+    rec = None
+    with open(path, "r") as records:
+        rec = records.readlines()
+    return rec[-1].split(",")[0]
+
+
+def get_date_range():
+    """Returns start and end date YY-MM-DD Formatted"""
+    start_date = get_last_date()
+    # start_date = "2022-07-30"
+    now = datetime.now()
```

Comment:
@gmyenni uncomment this `start_date line` to test with a few scenes. Could you try it anytime and let me know what you got on your side. You will need to create a credentials' file "usgs-pass.json" `{"username": "", "password": ""}` in the NDVI dir. You also need to install the deps `pip install landsatxplore`

Reply:
I have set start date as `start_date = "2022-07-30"`